### ISSUE TITLE
Stardew Valley searches for SDL2 in game path

### ISSUE THREAD

**Issue by Pixelnarium** [kujeger/flatpak-gog#25]
Commit https://github.com/kujeger/flatpak-gog/commit/cf8bdababbc403490d0429c7027e922031a25183 breaks Stardew Valley. 

`[ERROR] FATAL UNHANDLED EXCEPTION: System.DllNotFoundException: ./lib64/libSDL2-2.0.so.0`

My guess would be to create a configure-StardewValley file and link /usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0 to the lib64 directory. The problem is that the configure script is executed before the SDL2 removal command. So it should be moved to the top after the game installation?

There is also SDL2 dynapi https://hg.libsdl.org/SDL/file/default/docs/README-dynapi.md for that but I have not yet tested it.

**Comment by kujeger** [comment]
Have you tested setting LD_PRELOAD or LD_LIBRARY_PATH?

**Comment by V02460** [comment]
That should be a problem which will happen with other .NET games as well. The linking for the executable itself is just fine, there are other compat-shenanigans happening after launching it:

```
$ ldd StardewValley.bin.x86_64 
	linux-vdso.so.1 (0x00007ffe331cd000)
	libm.so.6 => /usr/lib/x86_64-linux-gnu/libm.so.6 (0x00007f183cf5f000)
	librt.so.1 => /usr/lib/x86_64-linux-gnu/librt.so.1 (0x00007f183cf55000)
	libdl.so.2 => /usr/lib/x86_64-linux-gnu/libdl.so.2 (0x00007f183cf4f000)
	libpthread.so.0 => /usr/lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f183cf2d000)
	libgcc_s.so.1 => /usr/lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f183cf13000)
	libc.so.6 => /usr/lib/x86_64-linux-gnu/libc.so.6 (0x00007f183cd4a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f183d0b8000)
```

The best way is probably to not remove the *.so file, but generally replace it with a symlink.

@kujeger I tried `$ LD_LIBRARY_PATH=/usr/lib/x86_64-linux-gnu ./StardewValley.bin.x86_64`, but it didn't work (same with `$ LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0 ./StardewValley.bin.x86_64`).

**Comment by V02460** [comment]
Regarding the Dynamic API: SDL2 was released in 2013 [[1]](https://en.wikipedia.org/wiki/Simple_DirectMedia_Layer#History) and the Dynamic API in 2014 [[2]](https://www.phoronix.com/scan.php?page=news_item&px=MTU2MzQ), about 5 months apart, so it should be pretty universal in games (and we have a very finite set of SDL2 games which don't have it).

Stardew Valley works with SDL_DYNAMIC_API set (but also without), I think there were some games where it did make a difference. If we can get those to work, using the Dynamic API would be the less hacky variant imo.

**Comment by kujeger** [comment]
Bit late, but I just pushed eddcb756ede62c68b1d653742fe4165c47471dd2 which replaces the deletion with SDL_DYNAMIC_API which ought to fix this. I tested a bunch of SDL2 game I own, and I haven't been able to find any that didn't work with it.

I don't have stardew valley though, so would appreciate a test.
I also pushed 63a75c828c31489a20430a5e7c87334aca9bb94a which gives a bit more power to configure scripts by making them the very last thing to run, so even if there are game-specific issues it should be easier to work around them now.
